go-controller/pkg/ovn/kubevirt.go

```diff
@@ -56,18 +64,300 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+	if err != nil {
+		return err
+	}
+	if !kubevirt.AllowPodBridgeNetworkLiveMigration(pod.Annotations) || isLiveMigrationLefover {
+		return nil
+	}
+	if err := oc.deleteDHCPOptions(pod); err != nil {
+		return err
+	}
+	if err := oc.deletePodEnrouting(pod); err != nil {
+		return err
+	}
+	if err := oc.cleanupVMSubnetsFromARPProxy(pod); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) addVMSubnetsToARPProxy(pod *kapi.Pod) error {
+	currentSwitchName, sourceSwitchName, err := oc.getSwitchNames(pod)
+	if err != nil {
+		return fmt.Errorf("failed retrieving pod related switch names: %v", err)
+	}
+
+	// The virtual machine ip does not belong to current switch
+	if currentSwitchName != sourceSwitchName {
+		subnets := oc.lsManager.GetSwitchSubnets(sourceSwitchName)
+		for _, subnet := range subnets {
+			if err := oc.ensureARPProxyContainsSubnet(currentSwitchName, subnet); err != nil {
+				return fmt.Errorf("failed ensuring current vm switch has subnet at ARP proxy")
+			}
+			if err := oc.ensureARPProxyContainsSubnet(sourceSwitchName, subnet); err != nil {
+				return fmt.Errorf("failed ensuring source vm switch has subnet at ARP proxy")
+			}
+		}
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) cleanupVMSubnetsFromARPProxy(pod *kapi.Pod) error {
+	currentSwitchName, sourceSwitchName, err := oc.getSwitchNames(pod)
+	if err != nil {
+		return fmt.Errorf("failed retrieving pod related switch names: %v", err)
+	}
+	podList, err := oc.client.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labels.Set{
+			kubevirt.SubnetSwitchNameLabel: sourceSwitchName,
+		}.String(),
+	})
+	if err != nil {
+		return err
+	}
+	hasMovedVirtLaunchersAtCurrentSwitch := false
+	hasMovedVirtLaunchers := false
+	for _, p := range podList.Items {
+		isLeftOver, err := kubevirt.PodIsLiveMigrationLeftOver(oc.client, &p)
 		if err != nil {
 			return err
 		}
-
-		if !isLiveMigrationLefover {
-			if err := oc.deleteDHCPOptions(pod); err != nil {
+		if isLeftOver {
+			continue
+		}
+		expectedSwitchName, err := oc.getExpectedSwitchName(&p)
+		if err != nil {
+			return err
+		}
+		if sourceSwitchName == expectedSwitchName {
+			continue
+		}
+		hasMovedVirtLaunchers = true
+		hasMovedVirtLaunchersAtCurrentSwitch = expectedSwitchName == currentSwitchName
+	}
+	subnets := oc.lsManager.GetSwitchSubnets(sourceSwitchName)
+	for _, subnet := range subnets {
+		if !hasMovedVirtLaunchers {
+			if err := oc.removeARPProxySubnet(sourceSwitchName, subnet); err != nil {
+				return err
+			}
+			if err := oc.removeARPProxySubnet(currentSwitchName, subnet); err != nil {
+				return err
+			}
+		} else if !hasMovedVirtLaunchersAtCurrentSwitch {
+			if err := oc.removeARPProxySubnet(currentSwitchName, subnet); err != nil {
 				return err
 			}
 		}
 	}
 	return nil
+
+}
+
+func (oc *DefaultNetworkController) enroutePodAddressesToNode(pod *kapi.Pod) error {
```

Comment:
I was hoping anywhere you touched LR policy or LR routes youd have a sync function in case we missed any events. WDYT? It might cost more CPU cycles but it at least will give us certainty about the state of the OVN DBs.

Reply:
Well, looks like I have to learn what the sync funcion is :-), I will add it.